### DISCUSSION TITLE
Remove Loaner Car

### DISCUSSION THREAD

**Discussion by natrlhy** [teslamate-org/teslamate#3074] [Q&A] [answered]
I just upgraded TeslaMate and see a loaner car show up. What's the right way to completely remove this vehicle and ID from my TeslaMate instance?

Thanks!

**Comment by cwanja** [comment]
#2719

**Comment by natrlhy** [comment, marked as answer]
Thanks @cwanja!
For reference I did the following:

SSH to `pi@teslamate`
In my case my ID I want to delete is `id=6`. Change based on your query results. Make sure to delete any `car_settings` ID's that don't map to a car you want to keep.
```
sudo -i
docker-compose exec database psql teslamate teslamate
SELECT id, vin FROM cars;
DELETE FROM cars WHERE id = 6;
SELECT * from car_settings;
SELECT id, settings_id, inserted_at, updated_at FROM cars;
DELETE FROM car_settings WHERE id = 3;
DELETE from charges WHERE charging_process_id in (select id from charging_processes where car_id = 6);
DELETE FROM charging_processes WHERE car_id = 6;
DELETE FROM drives WHERE car_id = 6;
DELETE FROM positions WHERE car_id = 6;
DELETE FROM states WHERE car_id = 6;
DELETE FROM updates WHERE car_id = 6;
```

Restart teslamate (do this after you've returned the loaner or have sold the car and it's not showing up in the Tesla App any longer otherwise the car will get added back to TeslaMate

```
docker-compose down
docker-compose up -d

**Reply by cwanja** [reply]
For sure, glad you got it working. I will make a PR to get docs updated until a native solution is added.

**Reply by natrlhy** [reply]
I've made some edits above to what I needed to do in my case. The `settings_id` didn't have a 1:1 mapping of the `car_id` to `settings_id`. So I had to do some extra checks to delete the right `car_settings` ID to fix getting the `Internal Server Error` on the http://teslamate:4000/settings page described  in https://github.com/adriankumpf/teslamate/issues/3032

Sample output from my use case:

```
teslamate=# select * from car_settings;
 id | suspend_min | suspend_after_idle_min | req_not_unlocked | free_supercharging | use_streaming_api
----+-------------+------------------------+------------------+--------------------+-------------------
  1 |          12 |                     10 | f                | f                  | t
  2 |          21 |                     15 | f                | f                  | t
  3 |          21 |                     15 | f                | f                  | t

teslamate=# select id, settings_id, inserted_at, updated_at from cars;
 id | settings_id |     inserted_at     |     updated_at
----+-------------+---------------------+---------------------
  6 |           3 | 2023-01-18 08:39:00 | 2023-01-18 10:17:31
  5 |           2 | 2022-02-06 20:58:30 | 2022-12-25 23:51:38
  1 |           1 | 2019-08-06 00:01:00 | 2022-12-29 03:32:05
```

**Reply by ednax** [reply]
I think you have to switch the delete of table _charging_processes_ and _charges_.

Because you select the table _charging_processes_ to delete the _charges_ - but that won't work, if you delete _charging_processes_ first.


sudo -i
docker-compose exec database psql teslamate teslamate
SELECT id, vin FROM cars;
DELETE FROM cars WHERE id = 6;
SELECT * from car_settings;
SELECT id, settings_id, inserted_at, updated_at FROM cars;
DELETE FROM car_settings WHERE id = 3;
DELETE from charges WHERE charging_process_id in (select id from charging_processes where car_id = 6);
DELETE FROM charging_processes WHERE car_id = 6;
DELETE FROM drives WHERE car_id = 6;
DELETE FROM positions WHERE car_id = 6;
DELETE FROM states WHERE car_id = 6;
DELETE FROM updates WHERE car_id = 6;

**Reply by marcbaier** [reply]
Where do I need to look in order to find out the specific ID that I need to delete ? I recently sold my old Tesla but Teslamate still is displaying the data for the old car and not for the new car. The Teslamate Settings page shows two tabs as well. One for the old car and one for the new car.

**Reply by natrlhy** [reply]
Hi @marcbaier 

You'll need to login to your teslamate containers. In my steps I do this:
```
ssh pi@teslamate (use the name or IP of your TeslaMate instance).
sudo -i
```
cd to the directory of your `docker-compose.yml` file

```
docker-compose exec database psql teslamate teslamate (replace `database` with whatever you named the container in your .yml fiile)
SELECT id, vin FROM cars; <--- this shows the `id` and `vin` of your cars Teslamate is logging:

 id |        vin
----+-------------------
  1 | myvin#1 here
  5 | myvin#2 here
 ```
Follow the other steps in my other post: https://github.com/adriankumpf/teslamate/discussions/3074#discussioncomment-4709108

**Comment by Harleyluhja** [comment]
Thanks a lot. I have made ist.
